pkg/execution/executor/executor.go

```diff
@@ -168,11 +168,8 @@ func WithLifecycleListeners(l ...execution.LifecycleListener) ExecutorOpt {
 	}
 }
 
-func WithStepLimits(limit uint) ExecutorOpt {
+func WithStepLimits(limit func(id state.Identifier) int) ExecutorOpt {
```

Comment:
shouldn't this be `uint` as well if the previous was an `uint`?

Reply:
its mainly for types, so that we don't need to continue to typecast everywhere.  we actively check for <= 0 in the validator so it's okay being an int, and is (i suppose) an easier type to work with.